### PR TITLE
Add `tags.scm` support

### PR DESCRIPTION
**Pull request by DavisVaughan**
Based on these examples
https://github.com/tree-sitter/tree-sitter-python/blob/master/queries/tags.scm
https://github.com/tree-sitter/tree-sitter-javascript/blob/master/queries/tags.scm
https://github.com/tree-sitter/tree-sitter-rust/blob/master/queries/tags.scm

And the Category/Tag table in this section
https://tree-sitter.github.io/tree-sitter/code-navigation-systems#examples

I used these
- `@definition.function`
- `@reference.call`

I don't _think_ we want any for class, interface, method, module, or implementation. The module one seems interesting, as it is somewhat related to package names, i.e. `dplyr::` or `library(dplyr)`, but I don't think it maps to anything, i.e. `definition.module` would be the creation of the "dplyr module", which is like `namespace dplyr {` in C++, but no equivalent in R, and there doesn't seem to be a `reference.module` oddly.

There also seems to be an undocumented one for constants but I don't think that works for us either
https://github.com/tree-sitter/tree-sitter-python/blob/a22761025cdac6c314b7e3aa48fb44fa9e594d6a/queries/tags.scm#L1

Let me know if you can think of something I'm missing!

### CODE REVIEW

File: queries/tags.scm [hunk 1]
```diff
@@ -0,0 +1,21 @@
+(binary_operator
+    lhs: (identifier) @name
+    operator: "<-"
+    rhs: (function_definition)
+) @definition.function
+
+(binary_operator
```
Comment (kevinushey) [inline queries/tags.scm:7]: Is it possible for us to detect function "aliases"? For example:

```
fn <- function(x) {}
alias <- fn
```

In other words, some way of checking for assignments where the rhs symbol might have already have a function definition? I'm guessing not since the tags are probably not stateful in any way, but figure I would ask.

File: queries/tags.scm [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    operator: "="
+    rhs: (function_definition)
+) @definition.function
+
+(call
+    function: (identifier) @name
+) @reference.call
+
+(call
+    function: (namespace_operator
```
Comment (kevinushey) [inline queries/tags.scm:18]: Is there any useful highlight we could apply to package names here?

File: test/tags/calls.R
[the remark this answers is not in the document]
Comment (kevinushey) [reply test/tags/calls.R:1]: Cool!

**Review by kevinushey — APPROVED** [review]
LGTM! Some other things that came to mind while reviewing; none of the below are totally necessary except _maybe_ the considerations on scoping of definitions.

---

Do we need to care / think about scoping, or is this handled for us by virtue of `@definition.function`? For example, in code like:

```
p1 <- function() {
  example <- function() {}
}

p2 <- function() {
  example <- function() {}
}
```

Will there be a way to disambiguate between p1's example and p2's example?

If so, should we also support some subset of common functions that create a scope, like `local()`?

---

Is it possible to highlight function parameters / arguments in some way? For example, so that in:

```
example <- function(var) {
  print(var)
}
```

we would have a way of noting that `var` is a parameter of the `example` function, so that go-to navigation would help note that this variable is a function parameter? Or is that out of scope?

---

Do we need to avoid detecting binary operators in formal function definitions? For example:

```
example <- function(var = foo <- 42) {
  var
  print(foo)
}
```

Such examples are fortunately very rare, but I think I've seen similar things in the R sources on occasion. (Might not be worth the effort unless someone notes / complains)

File: queries/tags.scm (hunk 1 above)
Comment (DavisVaughan) [reply queries/tags.scm:7]: Not that I am aware of, but I think that is mostly okay. I think this is going to be most useful when looking at package sources, where this is probably somewhat rare

File: queries/tags.scm (hunk 2 above)
Comment (DavisVaughan) [reply queries/tags.scm:18]: See original paragraph about `reference.module` https://github.com/r-lib/tree-sitter-r/pull/97#issue-2260037121

TLDR is that I don't think there is anything we can tag them with that github search would use (this isn't highlighting, this is for jump to definition and search)

File: package.json
Comment (DavisVaughan) [inline package.json:1]: > Do we need to avoid detecting binary operators in formal function definitions? 

Probably not worth looking into very much right now
